app/Http/Requests/RegisterUserRequest.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rules\Password;
+
+class RegisterUserRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     *
+     * @return bool
+     */
+    public function authorize()
+    {
+        return true;
+    }
+
+    /**
+     * Get the validation rules that apply to the request.
+     *
+     * @return array
+     */
+    public function rules()
+    {
+        return [
+            'username' => ['required', 'string', 'max:255'],
+            'email' => ['required', 'email', 'unique:App\Models\User,email', 'max:255'],
+            'password' => [
+                'required',
+                'string',
+                Password::min(8)
+                    ->letters()
+                    ->mixedCase()
+                    ->numbers()
+                    ->symbols()
+                    ->uncompromised()
+            ],
+            'confirmPassword' => ['required', 'string', 'same:password'],
+        ];
+    }
+
+    /**
+     * Get the error messages for the defined validation rules.
+     *
+     * @return array
+     */
+    public function messages()
+    {
+        return [
+            'username.required' => 'Please enter your username',
+            'username.max' => 'Please enter less than 255 characters',
+            'email.required' => 'Please enter your email',
```

Comment:
this is just to inform you that validation error messages are populated by default.

explicit definition is done only if we need to update the default validation message